components/raftstore/src/store/peer.rs

```diff
@@ -1801,6 +1808,15 @@ where
             };
         }
 
+        if !self.is_leader() {
```

Comment:
Should check whether there are persisted messages instead of whether it's leader.